docs/articles/persistence/custom-persistence-provider.md

```diff
@@ -0,0 +1,467 @@
+
+SQLite code example:
+
+[!code-csharp[LoadAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=LoadAsync "LoadAsync method implementation")]
+
+#### SaveAsync
+
+```c#
+Task SaveAsync(SnapshotMetadata metadata, object snapshot)
+```
+
+* **`metadata`**: Snapshot metadata
+* **`snapshot`**: The snapshot object instance to be stored
+
+This call is protected with a circuit-breaker
+
+Asynchronously saves a snapshot.
+
+##### Code Sample
+
+`InsertSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[InsertSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=InsertSnapshotSql "InsertSnapshotSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[SaveAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=SaveAsync "SaveAsync method implementation")]
+
+#### DeleteAsync
+
+```c#
+Task DeleteAsync(SnapshotMetadata metadata);
+```
+
+* **`metadata`**: Snapshot metadata
+
+This call is protected with a circuit-breaker
+
+Deletes the snapshot identified by by the provided `Metadata`
+
+##### Code Sample
+
+`DeleteSnapshotSql` in this example refers to this SQL query statement:
+
+[!code-csharp[DeleteSnapshotSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteSnapshotSql "DeleteSnapshotSql SQL Statement")]
+
+`DeleteSnapshotRangeSql` in this example refers to this SQL query statement:
+
+[!code-csharp[DeleteSnapshotRangeSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteSnapshotRangeSql "DeleteSnapshotRangeSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[DeleteAsync](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteAsync "DeleteAsync method implementation")]
+
+#### DeleteAsync
+
+```c#
+Task DeleteAsync(string persistenceId, SnapshotSelectionCriteria criteria);
+```
+
+* **`persistenceId`**: Identification of the persistent actor
+* **`criteria`**: Selection criteria for event deletion
+
+This call is protected with a circuit-breaker
+
+Deletes all snapshots matching the provided `criteria`
+
+##### Code Sample
+
+`DeleteSnapshotRangeSql` in this example refers to this SQL query statement:
+
+[!code-csharp[DeleteSnapshotRangeSql](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteSnapshotRangeSql "DeleteSnapshotRangeSql SQL Statement")]
+
+SQLite code example:
+
+[!code-csharp[DeleteAsync2](../../../src/examples/Akka.Persistence.Custom/Snapshot/SqliteSnapshotStore.cs?name=DeleteAsync2 "DeleteAsync method implementation")]
+
+### Detail Implementation
+
+#### Reading HOCON Settings
+
+There are some HOCON settings that are by default loaded by the snapshot store base class and these can be overriden in your HOCON settings. The minimum HOCON settings that need to be defined for your custom plugin are:
+
+```hocon
```

Comment:
What is the best option here?

Reply:
Nothing, HOCON is a proprietary format, there are no CSS equivalent, no support in DocFX

Reply:
You can still reference a HOCON file use a doc include - we do this in all of the default config docs

Reply:
I think the support is limited to code line range? and since HOCON is a nested structure in nature, linking to a line of code inside a HOCON is a bit meaningless because you would not get all of the parent property names before and can actually lead to more confusion by the reader

Reply:
you'd have to either include an entire stand-alone HOCON file (which works) or wrap around a C# format string using line numbers. Separate file is probably best.